### DISCUSSION TITLE
BigQuery REST

### DISCUSSION THREAD

**Discussion by AndyGUK** [Budibase/budibase#5101] [Help]
I've been trying to get my Budibase app to be able view data within my BigQuery table for what seems like months so I thought I might as well see if anyone had managed to do so successfully.

I can make successful requests using the REST data connector within Budibase, however, they're not permanent.  I don't have any experience in using REST APIs so forgive me if I'm missing something obvious!

I've used Postman to generate an access_token and value which seems to work. If anyone has any advice on how I:
1) Keep the connection alive so that the data is always available via Budibase
2) Transform the data requested via the API to be usable- at the moment, it comes through looking like this:
![image](https://user-images.githubusercontent.com/73335367/160166204-dc5dc474-14e6-4f25-b928-145b3086ac3a.png)

Any help at all will be massively appreciated!

**Comment by shogunpurple** [comment]
Hi @AndyGUK - if you are making a request to get credentials, and then using that in your query to bigquery, you will find the following docs helpful:
https://docs.budibase.com/docs/rest-variables#dynamic-variables

- Set up a REST query to fetch the access token etc (as you do now with postman)
- set up a dynamic variable for the tokens that come back from the response
- Use these in your calls to bigquery
This means that the values will be cached, so that every time you make your call to bigquery, if the credentials aren't present, budibase will go and fetch them again.

Let me know if you need any more guidance after reading the docs here.

**Reply by AndyGUK** [reply]
Hi @shogunpurple, thanks for sharing these docs. I've managed to have a read through and give it a go but unfortunate I'm missing something as it's not working.

I've tried adding all of the headers I'm using in Postman, all except the callback URL as I'm not sure how this works with budibase (and if using caddy could potentially impact the callback). 

I wonder if there's an easier way of keeping the auth alive. With my limited knowledge of BigQuery it seems like there are a couple of [different ways in which you can authenticate](https://cloud.google.com/bigquery/docs/authorization). 

Do you recon it would be easier to use another method of authentication or if there is perhaps a beginners guide somewhere that might be useful? I'm a complete novice at almost all of this so sorry for any negligence!
